apps/src/p5lab/spritelab/PoemBotLibrary.js

```diff
@@ -0,0 +1,214 @@
+import CoreLibrary from './CoreLibrary';
+import {commands as backgroundEffects} from './commands/poembot/backgroundEffects';
+import {commands as foregroundEffects} from './commands/poembot/foregroundEffects';
+
+const OUTER_MARGIN = 50;
+const LINE_HEIGHT = 50;
+const FONT_SIZE = 25;
+const TextType = {
+  BLANK: 'blank',
+  LITERAL: 'literal',
+  RANDOM: 'random'
+};
+const BLANK_TEXT = {value: '', type: TextType.BLANK};
+const PLAYSPACE_SIZE = 400;
+
+export default class PoemBotLibrary extends CoreLibrary {
+  constructor(p5) {
+    super(p5);
+    this.poem = {
+      title: '',
+      author: '',
+      lines: []
+    };
+    this.isVisible = false;
+    this.backgroundEffect = () => this.p5.background('white');
+    this.foregroundEffect = () => {};
+
+    this.commands = {
+      // Keep everything from Core Sprite Lab
+      ...this.commands,
+
+      // Override the draw loop
+      executeDrawLoopAndCallbacks() {
+        this.backgroundEffect();
+        this.runBehaviors();
+        this.runEvents();
+        this.p5.drawSprites();
+        this.drawPoem();
+        this.foregroundEffect();
+      },
+
+      // And add custom Poem Bot commands
+      textConcat(text1, text2) {
+        if (!text1) {
+          return text2;
+        }
+        if (!text2) {
+          return text1;
+        }
+        return [...text1, ...text2];
+      },
+
+      randomWord() {
+        // TODO: get curated random word list from Curriculum
+        const words = ['cat', 'dog', 'fish'];
+        const index = this.randomNumber(0, words.length - 1);
+        return words[index];
+      },
+
+      addLine(line) {
+        this.poem.lines.push(line || [BLANK_TEXT]);
+      },
+
+      setTitle(line) {
+        if (line) {
+          this.poem.title = line[0].value;
+        }
+      },
+
+      setAuthor(line) {
+        if (line) {
+          this.poem.author = line[0].value;
+        }
+      },
+
+      showPoem() {
+        this.isVisible = true;
+      },
+
+      hidePoem() {
+        this.isVisible = false;
+      },
+
+      setPoem(key) {
```

Comment:
nit / follow-up work: we should move these poems to a separate file (e.g., `poems.js`) so this function becomes a lot smaller. then, we could easily add/change poems within poems.js and access them within this library as `poem['wordsworth']`

Reply:
will do as follow up work (maybe once the actual poems are finalized)